Index: pytorch/cmake/Modules/FindMKL.cmake
===================================================================
diff --git cmake/Modules/FindMKL.cmake cmake/Modules/FindMKL.cmake
index ff24c97..797cb4c 100644
--- a/cmake/Modules/cmake/Modules/FindMKL.cmake
+++ b/cmake/Modules/cmake/Modules/FindMKL.cmake
@@ -113,8 +113,8 @@ ELSE(WIN32)
       SET(mklthreads "mkl_tbb_thread")
       SET(mklrtls "tbb")
     ELSE()
-      SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
-      SET(mklrtls "gomp" "iomp5")
+      SET(mklthreads "mkl_intel_thread" "mkl_gnu_thread")
+      SET(mklrtls "iomp5" "gomp")
     ENDIF()
     SET(mklifaces  "intel" "gf")
   ELSE(CMAKE_COMPILER_IS_GNUCC)
