From ac186a7b629043009abde727666c5bb307e7a76a Mon Sep 17 00:00:00 2001
From: conda-forge <conda-forge@users.noreply.github.com>
Date: Mon, 6 Nov 2023 13:46:00 -0500
Subject: [PATCH 02/13] Swap OpenMP search precedence to prefer Intel over GNU
---
 cmake/Modules/FindMKL.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: pytorch-2.12.0-rc3/cmake/Modules/FindMKL.cmake
===================================================================
--- pytorch-2.12.0-rc3.orig/cmake/Modules/FindMKL.cmake
+++ pytorch-2.12.0-rc3/cmake/Modules/FindMKL.cmake
@@ -113,8 +113,8 @@ ELSE(WIN32)
       SET(mklthreads "mkl_tbb_thread")
       SET(mklrtls "tbb")
     ELSE()
-      SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
-      SET(mklrtls "gomp" "iomp5")
+      SET(mklthreads "mkl_intel_thread" "mkl_gnu_thread")
+      SET(mklrtls "iomp5" "gomp")
     ENDIF()
     SET(mklifaces  "intel" "gf")
   ELSE(CMAKE_COMPILER_IS_GNUCC)
