Index: pytorch/cmake/Modules/FindMKL.cmake
===================================================================
--- pytorch.orig/cmake/Modules/FindMKL.cmake	2024-11-19 16:17:57.874552303 -0600
+++ pytorch/cmake/Modules/FindMKL.cmake	2024-11-19 16:18:29.915971379 -0600
@@ -108,8 +108,8 @@
       SET(mklthreads "mkl_tbb_thread")
       SET(mklrtls "tbb")
     ELSE()
-      SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
-      SET(mklrtls "gomp" "iomp5")
+      SET(mklthreads "mkl_intel_thread" "mkl_gnu_thread")
+      SET(mklrtls "iomp5" "gomp")
     ENDIF()
     SET(mklifaces  "intel" "gf")
   ELSE(CMAKE_COMPILER_IS_GNUCC)
