diff --git a/functorch/CMakeLists.txt b/functorch/CMakeLists.txt
index bdfa4bfe455..2baae3b2dee 100644
--- a/functorch/CMakeLists.txt
+++ b/functorch/CMakeLists.txt
@@ -17,7 +17,7 @@ target_compile_options(${PROJECT_NAME} PRIVATE ${TORCH_PYTHON_COMPILE_OPTIONS})
 target_compile_options_if_supported(${PROJECT_NAME} "-Wmissing-prototypes")
 target_compile_options_if_supported(${PROJECT_NAME} "-Werror=missing-prototypes")
 if(BUILD_LIBTORCHLESS)
-  target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIB} torch_python)
+  target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIB} torch_cpu c10 torch_python)
 else()
   # functorch cannot use the alias to build on windows
   target_link_libraries(${PROJECT_NAME} PRIVATE torch torch_python)
diff --git a/torch/CMakeLists.txt b/torch/CMakeLists.txt
index c74b45431c9..05b1f44502a 100644
--- a/torch/CMakeLists.txt
+++ b/torch/CMakeLists.txt
@@ -483,7 +483,7 @@ if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
           )
   # Pybind11 requires explicit linking of the torch_python library
   if(BUILD_LIBTORCHLESS)
-    target_link_libraries(nnapi_backend PRIVATE ${TORCH_LIB} torch_python pybind::pybind11)
+    target_link_libraries(nnapi_backend PRIVATE ${TORCH_LIB} torch_cpu c10 torch_python pybind::pybind11)
   else()
     target_link_libraries(nnapi_backend PRIVATE torch torch_python pybind::pybind11)
   endif()
diff --git a/torch/lib/libshm_windows/CMakeLists.txt b/torch/lib/libshm_windows/CMakeLists.txt
index df2a1064938..1d0c90e36a5 100644
--- a/torch/lib/libshm_windows/CMakeLists.txt
+++ b/torch/lib/libshm_windows/CMakeLists.txt
@@ -16,7 +16,7 @@ target_include_directories(shm PRIVATE
   ${CMAKE_CURRENT_SOURCE_DIR}
   )
 
-target_link_libraries(shm torch c10)
+target_link_libraries(shm torch c10 torch_cpu)
 
 
 install(TARGETS shm DESTINATION "${LIBSHM_INSTALL_LIB_SUBDIR}")
