From 441de792f785c7f31cbf78257a3302fdb35328ad Mon Sep 17 00:00:00 2001
From: Eric Lundby <Eric.Lundby@gmail.com>
Date: Fri, 13 Dec 2024 13:58:28 -0700
Subject: [PATCH] Remove optional pdb file on windows.

---
 c10/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/c10/CMakeLists.txt b/c10/CMakeLists.txt
index 80e172497d5..b794b528991 100644
--- a/c10/CMakeLists.txt
+++ b/c10/CMakeLists.txt
@@ -170,7 +170,3 @@ install(DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
         FILES_MATCHING PATTERN "*.h")
 install(FILES ${CMAKE_BINARY_DIR}/c10/macros/cmake_macros.h
         DESTINATION include/c10/macros)
-
-if(MSVC AND C10_BUILD_SHARED_LIBS)
-  install(FILES $<TARGET_PDB_FILE:c10> DESTINATION lib OPTIONAL)
-endif()
-- 
2.39.3 (Apple Git-146)

