diff --git a/cmake/Modules/FindCUB.cmake b/cmake/Modules/FindCUB.cmake
index 7c4647f3746..ff24c9787ad 100644
--- a/cmake/Modules/FindCUB.cmake
+++ b/cmake/Modules/FindCUB.cmake
@@ -3,7 +3,7 @@
 #  CUB_INCLUDE_DIRS - the CUB include directory
 
 find_path(CUB_INCLUDE_DIR
-        HINTS "${CUDAToolkit_INCLUDE_DIRS}"
+        HINTS "${CUDAToolkit_INCLUDE_DIRS}" "${CUDAToolkit_INCLUDE_DIRS}/cccl"
         NAMES cub/cub.cuh
         DOC "The directory where CUB includes reside"
 )
