From 3f989b1565148a96080ac14014d220e2b92002cc Mon Sep 17 00:00:00 2001
From: conda-forge <conda-forge@users.noreply.github.com>
Date: Thu, 6 Mar 2025 00:00:00 +0000
Subject: [PATCH 11/13] Add cccl subdirectory hint to FindCUB
---
 cmake/Modules/FindCUB.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Modules/FindCUB.cmake b/cmake/Modules/FindCUB.cmake
index 1b04d076749..02a650bbd18 100644
--- a/cmake/Modules/FindCUB.cmake
+++ b/cmake/Modules/FindCUB.cmake
@@ -3,7 +3,7 @@
 #  CUB_INCLUDE_DIRS - the CUB include directory
 
 find_path(CUB_INCLUDE_DIR
-        HINTS "${CUDAToolkit_INCLUDE_DIRS}"
+        HINTS "${CUDAToolkit_INCLUDE_DIRS}" "${CUDAToolkit_INCLUDE_DIRS}/cccl"
         NAMES cub/cub.cuh
         DOC "The directory where CUB includes reside"
 )
-- 
2.50.1 (Apple Git-155)

