diff --git a/c10/util/strong_type.h.orig b/c10/util/strong_type.h
index 1399c27c7d1..daf8a1804d2 100644
--- a/c10/util/strong_type.h.orig
+++ b/c10/util/strong_type.h
@@ -1604,12 +1604,6 @@ struct hash<::strong::type<T, Tag, M...>>
     return hash<T>::operator()(value_of(tt));
   }
 };
-template <typename T, typename Tag, typename ... M>
-struct is_arithmetic<::strong::type<T, Tag, M...>>
-  : is_base_of<::strong::arithmetic::modifier<::strong::type<T, Tag, M...>>,
-               ::strong::type<T, Tag, M...>>
-{
-};
 
 #if STRONG_HAS_STD_FORMAT
 template<typename T, typename Tag, typename... M, typename Char>
