--- a/c10/util/strong_type.h
+++ b/c10/util/strong_type.h
@@ -1604,13 +1604,6 @@ struct hash<::strong::type<T, Tag, M...>> {
   }
 };
 
-template <typename T, typename Tag, typename ... M>
-struct is_arithmetic<::strong::type<T, Tag, M...>>
-  : is_base_of<::strong::arithmetic::modifier<::strong::type<T, Tag, M...>>,
-               ::strong::type<T, Tag, M...>>
-{
-};
-
 #if STRONG_HAS_STD_FORMAT
 
 template<typename T, typename Tag, typename... M, typename Char>