--- cmake/MiscCheck.cmake.orig	2025-12-12 13:24:05.818413255 +0000
+++ cmake/MiscCheck.cmake	2025-12-12 13:24:40.106262760 +0000
@@ -100,8 +100,8 @@
 # due to protobuf.
 
 # ---[ Check if the compiler has SVE support.
-find_package(ARM) # checks SVE
-if(CXX_SVE_FOUND)
+# SVE disabled - GCC 14.3 compiler bug
+if(FALSE)
   message(STATUS "Compiler supports SVE extension. Will build perfkernels.")
   # Also see CMakeLists.txt under caffe2/perfkernels.
   add_compile_definitions(CAFFE2_PERF_WITH_SVE=1)
