--- caffe2/CMakeLists.txt.orig	2026-01-16 12:11:10.447162648 +0000
+++ caffe2/CMakeLists.txt	2026-01-16 12:10:09.240321579 +0000
@@ -1926,3 +1926,8 @@
   endforeach()
 endif()
 endif()
+
+# FIX: Force OneDNN headers for CUDA build (Fixes undefined symbol: fuseGraph)
+if(USE_CUDA)
+  target_include_directories(torch_cuda PRIVATE "\${PROJECT_SOURCE_DIR}/third_party/ideep/mkl-dnn/include")
+endif()
\ No newline at end of file
