diff --git a/torch/csrc/utils/python_arg_parser.h b/torch/csrc/utils/python_arg_parser.h
index 6a05807e5a..238c599948 100644
--- a/torch/csrc/utils/python_arg_parser.h
+++ b/torch/csrc/utils/python_arg_parser.h
@@ -120,7 +120,7 @@ private:
 
   std::vector<FunctionSignature> signatures_;
   std::string function_name;
-  ssize_t max_args;
+  Py_ssize_t max_args;
   bool traceable;
 };
 
@@ -135,9 +135,9 @@ struct PYBIND11_EXPORT FunctionSignature {
   std::string name;
   std::vector<FunctionParameter> params;
   std::vector<py::handle> overloaded_args;
-  ssize_t min_args;
-  ssize_t max_args;
-  ssize_t max_pos_args;
+  Py_ssize_t min_args;
+  Py_ssize_t max_args;
+  Py_ssize_t max_pos_args;
   int index;
   bool hidden;
   bool deprecated;
