Index: pytorch/third_party/fbgemm/third_party/asmjit/src/asmjit/core/environment.h
===================================================================
--- pytorch.orig/third_party/fbgemm/third_party/asmjit/src/asmjit/core/environment.h
+++ pytorch/third_party/fbgemm/third_party/asmjit/src/asmjit/core/environment.h
@@ -101,7 +101,7 @@ enum class Platform : uint8_t {
     kDragonFlyBSD
 #elif defined(__HAIKU__)
     kHaiku
-#elif defined(__APPLE__) && TARGET_OS_OSX
+#elif defined(__APPLE__) && TARGET_OS_MAC
     kOSX
 #elif defined(__APPLE__) && TARGET_OS_TV
     kTVOS
Index: pytorch/third_party/fbgemm/third_party/asmjit/src/asmjit/core/virtmem.cpp
===================================================================
--- pytorch.orig/third_party/fbgemm/third_party/asmjit/src/asmjit/core/virtmem.cpp
+++ pytorch/third_party/fbgemm/third_party/asmjit/src/asmjit/core/virtmem.cpp
@@ -28,7 +28,7 @@
   #if defined(__APPLE__)
     #include <pthread.h>
     #include <TargetConditionals.h>
-    #if TARGET_OS_OSX
+    #if TARGET_OS_MAC
       #include <sys/utsname.h>
       #include <libkern/OSCacheControl.h> // sys_icache_invalidate().
     #endif
@@ -413,7 +413,7 @@ static int mmProtFromMemoryFlags(MemoryF
 // Detects whether the current process is hardened, which means that pages that have WRITE and EXECUTABLE flags cannot
 // be allocated without MAP_JIT flag.
 static inline bool hasHardenedRuntimeMacOS() noexcept {
-#if TARGET_OS_OSX && ASMJIT_ARCH_ARM >= 64
+#if TARGET_OS_MAC && ASMJIT_ARCH_ARM >= 64
   // MacOS on AArch64 has always hardened runtime enabled.
   return true;
 #else
@@ -445,11 +445,11 @@ static inline bool hasHardenedRuntimeMac
 }
 
 static inline bool hasMapJitSupportMacOS() noexcept {
-#if TARGET_OS_OSX && ASMJIT_ARCH_ARM >= 64
+#if TARGET_OS_MAC && ASMJIT_ARCH_ARM >= 64
   // MacOS for 64-bit AArch architecture always uses hardened runtime. Some documentation can be found here:
   //   - https://developer.apple.com/documentation/apple_silicon/porting_just-in-time_compilers_to_apple_silicon
   return true;
-#elif TARGET_OS_OSX
+#elif TARGET_OS_MAC
   // MAP_JIT flag required to run unsigned JIT code is only supported by kernel version 10.14+ (Mojave) and IOS.
   static std::atomic<uint32_t> globalVersion;
 
