diff -urN a/third_party/fbgemm/third_party/asmjit/src/asmjit/core/environment.h b/third_party/fbgemm/third_party/asmjit/src/asmjit/core/environment.h
--- a/third_party/fbgemm/third_party/asmjit/src/asmjit/core/environment.h	2022-02-16 13:33:31.000000000 +0300
+++ b/third_party/fbgemm/third_party/asmjit/src/asmjit/core/environment.h	2022-02-16 15:16:14.000000000 +0300
@@ -278,7 +278,7 @@
     kPlatformDragonFlyBSD
 #elif defined(__HAIKU__)
     kPlatformHaiku
-#elif defined(__APPLE__) && TARGET_OS_OSX
+#elif defined(__APPLE__) && TARGET_OS_MAC
     kPlatformOSX
 #elif defined(__APPLE__) && TARGET_OS_TV
     kPlatformTVOS
diff -urN a/third_party/fbgemm/third_party/asmjit/src/asmjit/core/virtmem.cpp b/third_party/fbgemm/third_party/asmjit/src/asmjit/core/virtmem.cpp
--- a/third_party/fbgemm/third_party/asmjit/src/asmjit/core/virtmem.cpp	2022-02-16 13:33:31.000000000 +0300
+++ b/third_party/fbgemm/third_party/asmjit/src/asmjit/core/virtmem.cpp	2022-02-16 15:16:25.000000000 +0300
@@ -45,7 +45,7 @@
   // Apple recently introduced MAP_JIT flag, which we want to use.
   #if defined(__APPLE__)
     #include <TargetConditionals.h>
-    #if TARGET_OS_OSX
+    #if TARGET_OS_MAC
       #include <sys/utsname.h>
     #endif
     // Older SDK doesn't define `MAP_JIT`.
@@ -428,7 +428,7 @@
 // MAP_JIT flag required to run unsigned JIT code is only supported by kernel
 // version 10.14+ (Mojave) and IOS.
 static ASMJIT_INLINE bool VirtMem_hasMapJitSupport() noexcept {
-#if TARGET_OS_OSX
+#if TARGET_OS_MAC
   static volatile int globalVersion;
 
   int ver = globalVersion;
