Index: pytorch/torch/utils/cpp_extension.py
===================================================================
--- pytorch.orig/torch/utils/cpp_extension.py
+++ pytorch/torch/utils/cpp_extension.py
@@ -287,6 +287,8 @@ def check_compiler_ok_for_platform(compi
     '''
     if IS_WINDOWS:
         return True
+    if os.sep in compiler:
+        return os.path.exists(compiler)
     which = subprocess.check_output(['which', compiler], stderr=subprocess.STDOUT)
     # Use os.path.realpath to resolve any symlinks, in particular from 'c++' to e.g. 'g++'.
     compiler_path = os.path.realpath(which.decode(*SUBPROCESS_DECODE_ARGS).strip())
