Index: pytorch/CMakeLists.txt
===================================================================
--- pytorch.orig/CMakeLists.txt
+++ pytorch/CMakeLists.txt
@@ -458,13 +458,6 @@ option(
   "Master flag to build Lite Interpreter with tracing build option"
   OFF)
 
-# This is a fix for a rare build issue on Ubuntu:
-# symbol lookup error: miniconda3/envs/pytorch-py3.7/lib/libmkl_intel_lp64.so: undefined symbol: mkl_blas_dsyrk
-# https://software.intel.com/en-us/articles/symbol-lookup-error-when-linking-intel-mkl-with-gcc-on-ubuntu
-if(LINUX)
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-as-needed")
-endif()
-
 if(MSVC)
   set(CMAKE_NINJA_CMCLDEPS_RC OFF)
   foreach(flag_var
