--- test/run_test.py	2025-10-02 13:39:34.543337955 +0100
+++ "test/run_test copy.py"	2025-10-02 16:02:58.515409486 +0100
@@ -557,31 +557,31 @@
 # Test files that should always be run serially with other test files,
 # but it's okay if the tests inside them are run in parallel with each other.
 CI_SERIAL_LIST = [
-    "test_nn",
-    "test_fake_tensor",
-    "test_cpp_api_parity",
-    "test_reductions",
-    "test_fx_backends",
-    "test_cpp_extensions_jit",
-    "test_torch",
-    "test_tensor_creation_ops",
-    "test_dispatch",
-    "test_python_dispatch",  # torch.library creation and deletion must be serialized
-    "test_spectral_ops",  # Cause CUDA illegal memory access https://github.com/pytorch/pytorch/issues/88916
-    "nn/test_pooling",
-    "nn/test_convolution",  # Doesn't respect set_per_process_memory_fraction, results in OOM for other tests in slow gradcheck
-    "distributions/test_distributions",
-    "test_fx",  # gets SIGKILL
-    "functorch/test_memory_efficient_fusion",  # Cause CUDA OOM on ROCm
-    "test_utils",  # OOM
-    "test_sort_and_select",  # OOM
-    "test_backward_compatible_arguments",  # OOM
-    "test_autocast",  # OOM
-    "test_native_mha",  # OOM
-    "test_module_hooks",  # OOM
-    "inductor/test_max_autotune",
-    "inductor/test_cutlass_backend",  # slow due to many nvcc compilation steps,
-    "inductor/test_flex_attention",  # OOM
+    #"test_nn",
+    #"test_fake_tensor", 
+    #"test_cpp_api_parity",
+    #"test_reductions",
+    #"test_fx_backends",
+    #"test_cpp_extensions_jit",
+    #"test_torch",
+    #"test_tensor_creation_ops",
+    #"test_dispatch",
+    #"test_python_dispatch",  # torch.library creation and deletion must be serialized
+    #"test_spectral_ops",  # Cause CUDA illegal memory access https://github.com/pytorch/pytorch/issues/88916
+    #"nn/test_pooling",
+    #"nn/test_convolution",  # Doesn't respect set_per_process_memory_fraction, results in OOM for other tests in slow gradcheck
+    #"distributions/test_distributions",
+    #"test_fx",  # gets SIGKILL
+    #"functorch/test_memory_efficient_fusion",  # Cause CUDA OOM on ROCm
+    #"test_utils",  # OOM
+    #"test_sort_and_select",  # OOM
+    #"test_backward_compatible_arguments",  # OOM
+    #"test_autocast",  # OOM
+    #"test_native_mha",  # OOM
+    #"test_module_hooks",  # OOM
+    #"inductor/test_max_autotune",
+    #"inductor/test_cutlass_backend",  # slow due to many nvcc compilation steps,
+    #"inductor/test_flex_attention",  # OOM
 ]
 # A subset of onnx tests that cannot run in parallel due to high memory usage.
 ONNX_SERIAL_LIST = [
@@ -594,15 +594,8 @@
 
 # A subset of our TEST list that validates PyTorch's ops, modules, and autograd function as expected
 CORE_TEST_LIST = [
-    "test_autograd",
-    "test_autograd_fallback",
-    "test_modules",
-    "test_nn",
     "test_ops",
     "test_ops_gradients",
-    "test_ops_fwd_gradients",
-    "test_ops_jit",
-    "test_torch",
 ]
 
 
@@ -1588,9 +1581,9 @@
         "--include",
         nargs="+",
         choices=TestChoices(TESTS),
-        default=TESTS,
+        default=CORE_TEST_LIST,
         metavar="TESTS",
-        help="select a set of tests to include (defaults to ALL tests)."
+        help="select a set of tests to include (defaults to CORE_TEST_LIST: test_ops and test_ops_gradients)."
         " tests must be a part of the TESTS list defined in run_test.py",
     )
     parser.add_argument(
