Index: pytorch/third_party/XNNPACK/src/jit/memory.c
===================================================================
--- pytorch.orig/third_party/XNNPACK/src/jit/memory.c
+++ pytorch/third_party/XNNPACK/src/jit/memory.c
@@ -32,6 +32,9 @@ enum xnn_status xnn_allocate_code_memory
     return xnn_status_out_of_memory;
   }
 #else
+#ifndef MAP_ANONYMOUS
+#define MAP_ANONYMOUS MAP_ANON
+#endif
   void* p = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
   if (p == MAP_FAILED) {
     xnn_log_error("failed to allocate %zu bytes for JIT code buffer, error code: %d", size, errno);
